core/src/main/java/org/mskcc/cbio/portal/dao/JdbcUtil.java

```diff
@@ -56,7 +56,7 @@
      */
     public static DataSource getDataSource() {
         if (dataSource == null) {
-            dataSource = new JdbcDataSource();
```

Comment:
We need to be careful here. There is code which wires beans appropriately during spring startup for determination of the data source based on portal.properties and application-context files. We use a shared data source within tomcat when running multiple instances of cBioPortal within a single tomcat application. It seems like this code change may hardwire a connection pool through the apache BasicDataSource library onto each running tomcat web application.
We will do some testing of this. Maybe @n1zea144 can comment on this code change.

Reply:
good points, if you need to run multiple portal instances in a single tomcat container. Will revert the change.

Reply:
@sheridancbio , only with one change made to JdbcDataSource constructor, to replace hard code mysql driver class name with value read from portal.properties file as the following:
from
> this.setDriverClassName("com.mysql.jdbc.Driver");

to

> String mysqlDriverClassName = dbProperties.getDbDriverClassName();
> this.setDriverClassName(mysqlDriverClassName);

Reply:
reminder to self : let's test this code in tomcat